examples/fluid/dam_break_2phase_2d.jl

```diff
@@ -0,0 +1,96 @@
+
+# ==========================================================================================
+# ==== Experiment Setup
+gravity = 9.81
+tspan = (0.0, 2.0)
+
+# Numerical settings
+smoothing_length = 3.5 * fluid_particle_spacing
+sound_speed = 100.0
+# when using the Ideal gas equation
+# sound_speed = 343.0
+
+# physical values
+nu_water = 8.9E-7
+nu_air = 1.544E-5
+nu_ratio = nu_water / nu_air
+
+nu_sim_air = 0.02 * smoothing_length * sound_speed
+nu_sim_water = nu_ratio * nu_sim_air
+
+air_viscosity = ViscosityMorris(nu=nu_sim_air)
+water_viscosity = ViscosityMorris(nu=nu_sim_water)
+
+trixi_include(@__MODULE__, joinpath(examples_dir(), "fluid", "dam_break_2d.jl"),
+              sol=nothing, fluid_particle_spacing=fluid_particle_spacing,
+              viscosity=water_viscosity, smoothing_length=smoothing_length,
+              gravity=gravity, tspan=tspan, density_diffusion=nothing,
+              sound_speed=sound_speed, exponent=7,
+              tank_size=(floor(5.366 * H / fluid_particle_spacing) * fluid_particle_spacing,
+                         2.6 * H))
+
+# ==========================================================================================
+# ==== Setup air_system layer
+
+air_size = (tank_size[1], 1.5 * H)
+air_size2 = (tank_size[1] - W, H)
+air_density = 1.0
+
+# Air above the initial water
+air_system = RectangularShape(fluid_particle_spacing,
+                              round.(Int, air_size ./ fluid_particle_spacing),
+                              zeros(length(air_size)), density=air_density)
+
+# Air for the rest of the empty volume
+air_system2 = RectangularShape(fluid_particle_spacing,
+                               round.(Int, air_size2 ./ fluid_particle_spacing),
+                               (W, 0.0), density=air_density)
+
+# move on top of the water
+for i in axes(air_system.coordinates, 2)
+    air_system.coordinates[:, i] .+= [0.0, H]
+end
+
+air_system = union(air_system, air_system2)
+
+air_system_system = WeaklyCompressibleSPHSystem(air_system, fluid_density_calculator,
+                                                StateEquationCole(; sound_speed,
+                                                                  reference_density=air_density,
+                                                                  exponent=1,
+                                                                  clip_negative_pressure=false,
+                                                                  background_pressure=0),
+                                                smoothing_kernel, smoothing_length,
+                                                viscosity=air_viscosity,
+                                                acceleration=(0.0, -gravity))
+
+# air_system_system = WeaklyCompressibleSPHSystem(air_system, fluid_density_calculator,
+#                                          StateEquationIdealGas(; sound_speed, reference_density=air_density, gamma=1.4, background_pressure=0),
```

Comment:
This here is identical apart from the state equation, isn't it? Can you just define the state equation before the system and add the ideal gas equation commented out?